glib/src/subclass/types.rs

```diff
@@ -233,23 +234,48 @@ impl TypeData {
         self.parent_class
     }
 
-    /// Returns a pointer to the interface implementation specific data.
+    /// Returns a pointer to the class implementation specific data.
     ///
-    /// This is used for interface implementations to store additional data.
-    pub fn get_interface_data(&self, type_: ffi::GType) -> ffi::gpointer {
+    /// This is used for class implementations to store additional data.
+    pub fn get_class_data<T: Any + Send + Sync + 'static>(&self, type_: ffi::GType) -> Option<&T> {
         unsafe {
-            if self.interface_data.is_null() {
-                return ptr::null_mut();
+            if self.class_data.is_null() {
+                return None;
             }
 
-            for &(t, p) in &(*self.interface_data) {
-                if t == type_ {
-                    return p;
-                }
-            }
+            (*(self.class_data as *const HashMap<ffi::GType, T>)).get(&type_)
```

Comment:
That won't work. You need to `get()` it with the `Box<_>` type from above and then call `downcast_ref()` on it to get the `&T` out of it.

Reply:
Also everywhere else. You need to handle the hashmap with its original type and make use of the `Any` API